test/integration/authorise_application_test.rb

```diff
@@ -6,22 +6,44 @@ class AuthoriseApplicationTest < ActionDispatch::IntegrationTest
     @user = create(:user)
   end
 
-  should "not confirm the authorisation for a signed-out user" do
+  should "not confirm the authorisation until the user signs in" do
     visit "/oauth/authorize?response_type=code&client_id=#{@app.uid}&redirect_uri=#{@app.redirect_uri}"
     assert_response_contains("You need to sign in")
-    signin(@user)
 
-    refute Doorkeeper::AccessGrant.find_by(resource_owner_id: @user.id)
```

Comment:
This was mistakenly switched to a `refute` in b19763283d94a545b140958bb4a5d99db00a8f19 - I think the test was therefore wrong - so this is switching it back. You are right though, we should have a `refute` before the `signin(@user)` line.